src/fsharp/pars.fsy

```diff
@@ -895,7 +894,13 @@ tyconSpfn:
         $3 lhsm $1 (Some mEquals) }
   | typeNameInfo  opt_classSpfn       
       { let mWithKwd, members = $2
-        SynTypeDefnSig($1, None, SynTypeDefnSigRepr.Simple (SynTypeDefnSimpleRepr.None (lhs parseState), lhs parseState), mWithKwd, members, lhs parseState) }
+        let m =
+            match members with
+            | [] -> lhs parseState
```

Comment:
Wouldn't be safer to use `typeNameInfo` and `mWithKwd` ranges instead?

There's a chance there're no members produced due to recovering from an error inside `opt_classSpfn`, but the parse state may still include further tokens, leading to situation where one token belongs to two different declarations, similar to the example in https://github.com/dotnet/fsharp/pull/12224.